src/app/inventory/store/d2-item-factory.service.ts

```diff
@@ -648,7 +649,28 @@ function getClassTypeNameLocalized(defs: D2ManifestDefinitions, type: DestinyCla
     return t('Loadouts.Any');
   }
 }
+function getSeason(item) {
+  let sourceToSeason;
+  if (item.classified) {
+    return D2CalculatedSeason;
+  }
+  if (
+    D2SeasonToSource.categoryBlacklist.filter((itemHash) =>
+      item.itemCategoryHashes.includes(itemHash)
+    ).length ||
+    !item.itemCategoryHashes.length ||
+    item.typeName === 'Unknown'
+  ) {
+    return -1;
+  }
+  Object.keys(D2SeasonToSource.seasons).forEach((season) => {
```

Comment:
Not important, but you could use a standard `for..of` loop here and early return.

Reply:
`for..of`  says `D2SeasonToSource.seasons` is not iterable

and early out causes a bunch of items to be misidentified, I'm not sure why.